lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -13,3 +13,71 @@
 remove_action( 'wp_body_open', 'wp_global_styles_render_svg_filters' );
 remove_action( 'in_admin_header', 'wp_global_styles_render_svg_filters' );
 
+/**
+ * Collect the block editor assets that need to be loaded into the editor's iframe.
+ *
+ * @since 6.0.0
+ * @access private
+ *
+ * @global string $pagenow The filename of the current screen.
+ *
+ * @return array {
+ *     The block editor assets.
+ *
+ *     @type string|false $styles  String containing the HTML for styles.
+ *     @type string|false $scripts String containing the HTML for scripts.
+ * }
+ */
+function _wp_get_iframed_editor_assets__63() {
+	global $wp_styles, $wp_scripts;
+
+	// Keep track of the styles and scripts instance to restore later.
+	$current_wp_styles  = $wp_styles;
+	$current_wp_scripts = $wp_scripts;
+
+	// Create new instances to collect the assets.
+	$wp_styles  = new WP_Styles();
+	$wp_scripts = new WP_Scripts();
+
+	// Register all currently registered styles and scripts. The actions that
+	// follow enqueue assets, but don't necessarily register them.
+	$wp_styles->registered  = $current_wp_styles->registered;
+	$wp_scripts->registered = $current_wp_scripts->registered;
+
+	// We don't want to load EDITOR scripts and styles in the iframe, only
+	// assets for the content.
+	add_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );
+	do_action( 'enqueue_block_assets' );
+	remove_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );
+
+	wp_enqueue_style( 'wp-block-editor-content' );
+	wp_enqueue_script( 'wp-polyfill' );
+
+	ob_start();
+	wp_print_styles();
+	$styles = ob_get_clean();
+
+	ob_start();
+	wp_print_head_scripts();
+	wp_print_footer_scripts();
+	$scripts = ob_get_clean();
+
+	// Restore the original instances.
+	$wp_styles  = $current_wp_styles;
+	$wp_scripts = $current_wp_scripts;
+
+	return array(
+		'styles'  => $styles,
```

Comment:
Documenting how we fill this array [in core](https://github.com/WordPress/wordpress-develop/blob/d05b583c99d91596817266517e69e2572b74efdf/src/wp-includes/block-editor.php#L298):

- `styles`
  - `wp-polyfill`
  - for all registered blocks: the `$script_handles` (`script` asset from `block.json` as well as any other registered by other means)
  - all the dependencies of the previous two items (automatically resolved by [wp_scripts->do_items](https://github.com/WordPress/wordpress-develop/blob/d05b583c99d91596817266517e69e2572b74efdf/src/wp-includes/class-wp-dependencies.php#L127))
- `scripts`
  - `wp-edit-blocks`
  - `wp-block-library-theme` if theme supports `wp-block-styles` but hasn't provided any
  - `wp-widgets` and `wp-edit-widgets` for the widgets editor
  - for all registered blocks: the `$style_handles` and `$editor_style_handles` (`style` and `editorStyles` assets from `block.json` respectively, as well as any other registered by other means)
  - all the dependencies of the previous items (automatically resolved by [wp_styles->do_items](https://github.com/WordPress/wordpress-develop/blob/d05b583c99d91596817266517e69e2572b74efdf/src/wp-includes/class-wp-dependencies.php#L127)
  - `wp-reset-editor-styles`: avoid enqueuing it in the iframe [by adding it to the done queue](https://github.com/WordPress/wordpress-develop/blob/d05b583c99d91596817266517e69e2572b74efdf/src/wp-includes/class-wp-dependencies.php#L130), so it's ignored

Reply:
All of this should be covered, except the widgets stuff as the iframe is not being used there right now. `wp-reset-editor-styles` is not added.

Reply:
I've looked at the assets loaded in the iframe for the post editor using the TwentyTwentyThree theme.

Extra things this PR has that are not present in trunk:

- The assets added via `enqueue_block_assets`. This is correct. I've used [this test plugin](https://github.com/oandregal/understanding-gutenberg/blob/trunk/hooks-assets/hooks-assets.php): 
  - `enqueue_block_assets_style_body` is for styles without a `.wp-block` class
  - `enqueue_block_assets_style_wp_block` is for styles with `.wp-block` class
- The inline styles for emojis, which are not present in trunk. Are these necessary?

Things this PR misses that are present in trunk:

- `wp-reusable-blocks`. I'm a bit surprised about this. It should have been present because it's a [dependency](https://github.com/WordPress/gutenberg/blob/trunk/lib/client-assets.php#L320) of `wp-edit-blocks`, which is enqueued. The same issue happens with `wp-components` and we had to [declare it as a dependency of `wp-block-editor-content`](https://github.com/WordPress/gutenberg/pull/49655/files#r1176151529) for it to be enqueued. It sounds like all dependencies of `wp-edit-blocks` are missing. Any ideas why would this happen? :thinking: 
- `wp-fonts-local`. It sounds like these are fonts defined by the theme.

| Post Editor Assets                                           | Trunk | This PR |
| ------------------------------------------------ | ----- | ---- |
| `html.height.auto, body.margin.0`  (inline)      | x     | x    |
| `img.wp-smiley, img.emoji` (inline)              |       | x    |
| `dashicons-css`                                  | x     | x    |
| `wp-components-css`                              | x     | x    |
| `wp-block-editor-content-css`                    | x     | x    |
| `wp-block-library-css`                           | x     | x    |
| `wp-edit-blocks-css`                             | x     | x    |
| `wp-reusable-blocks-css`                         | x     |      |
| `wp-fonts-local` (inline)                        | x     |      |
| `enqueue_block_assets_style_body-css`            |       | x    |
| `enqueue_block_assets_style_wp_block-css`        | x     | x    |
| `enqueue_block_editor_assets_style_wp_block-css` | x     | x    |
| `wp-inert-polyfill-js`                           | x     | x    |
| `enqueue_block_assets_script_console-js`         |       | x    |
| `enqueue_block_assets_wp_api-js`                 |       | x    |
| `wp-polyfill-inert-js`                           | x     | x    |
| `regenerator-runtime-js`                         | x     | x    |
| `wp-polyfill-js`                                 | x     | x    |

Reply:
`wp-reusable-blocks` is not there because `wp-edit-blocks` is added through the compat layer. We need to change `wp-edit-blocks` in a follow up PR so it's not added anymore. `wp-reusable-blocks` does not contain any styles relevant to the editor content.

`wp-fonts-local` this is not present in core either. Looks like these are added in the 6.2, but not added to WP core. I don't know why. Since they are not present in core I don't think it's a big deal (see lib/compat/wordpress-6.2/script-loader.php)